skyportal/handlers/api/obj.py

```diff
@@ -0,0 +1,70 @@
+from baselayer.app.access import auth_or_token
+from ..base import BaseHandler
+from ...models import Candidate, DBSession, Filter, Group, Obj, Photometry, Source
+
+
+class ObjShortHandler(BaseHandler):
+    @auth_or_token
+    def get(self, obj_id=None):
+        """
+        ---
+        single:
+          description: Check if an Obj exists (either a Candidate or a Source)
+          parameters:
+            - in: path
+              name: obj_id
+              required: true
+              schema:
+                type: string
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: SingleObj
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+        user_group_ids = [g.id for g in self.associated_user_object.accessible_groups]
+        num_s = (
+            DBSession()
+            .query(Source)
+            .filter(Source.obj_id == obj_id)
+            .filter(Source.group_id.in_(user_group_ids))
+            .count()
+        )
+        num_c = (
+            DBSession()
+            .query(Candidate)
+            .filter(Candidate.obj_id == obj_id)
+            .filter(
+                Candidate.filter_id.in_(
+                    DBSession.query(Filter.id).filter(
+                        Filter.group_id.in_(user_group_ids)
+                    )
+                )
+            )
+            .count()
+        )
+        if num_s > 0 or num_c > 0:
+            return self.success(
+                data={"id": obj_id, "is_candidate": num_c > 0, "is_source": num_s > 0}
```

Comment:
Do you use the 2 last values  on the Kowalski side? If not, a lot of this could be replaced with

https://github.com/skyportal/skyportal/blob/master/skyportal/models.py#L891

Reply:
Firstly, I do; secondly, that stuff does a ton of unnecessary things and is therefore significantly slower (tried using that originally)